tests/integration/datadict/submission/test_endpoints.py

```diff
@@ -971,3 +971,93 @@ def test_duplicate_submission(app, pg_driver, cgci_blgsp, submitter):
 
     with pg_driver.session_scope():
         assert pg_driver.nodes(md.Experiment).count() == 1
+
+
+def test_number_transform(client, pg_driver, cgci_blgsp, submitter):
+    """
+    Test that a json float is correctly converted to an int
```

Comment:
let's put more detail in the test name - `test_zero_decimal_float`

and the description isn't right, let's add details too. we are testing that float values with a zero decimal are accepted by Sheepdog for properties of type "number" even if they look like integers. And you can mention we are testing with a TSV because the str values from the TSVs are casted to the proper type by Sheepdog